src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,16 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="ConstructorParameterShouldMatchFieldName" xml:space="preserve">
+    <value>In the constructor of '{0}', change the name of parameter '{1}', to match the referenced field '{2}'</value>
```

Comment:
Sounds good, especially considering that the `JsonConstructor` attribute also uses "bound" in its exception message. It may be useful to have a containing class name in the message for some cases, but I'm not completely sure. I'll change it to the suggested one.

Reply:
> It may be useful to have a containing class name in the message for some cases, but I'm not completely sure.

Including the class name or word `constructor` in the message sounds reasonable to me

Reply:
Thanks, added.